### PR TITLE
add jasper/2.0.16

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **jasper/2.0.16**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jasper/all/conanfile.py [hunk 1]
```diff
@@ -37,25 +45,22 @@ def source(self):
...
+        self._cmake.definitions["JAS_ENABLE_PROGRAMS"] = False
+        self._cmake.definitions["JAS_ENABLE_SHARED"] = self.options.shared
+        self._cmake.definitions["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
+        self._cmake.definitions["JAS_ENABLE_OPENGL"] = False
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
 
     def build(self):
-        self._patch_cmake()
+        os.rename(os.path.join(self._source_subfolder, "CMakeLists.txt"),
```
Comment (Croydon) [inline recipes/jasper/all/conanfile.py]: Since you have added a CMake wrapper it shouldn't be necessary to rename the original CMake file

The original file is in `source_subfolder`, the wrapper one directory above

Comment (SpaceIm) [reply recipes/jasper/all/conanfile.py]: Tried that the first time, but I had an error related to include(InSourceBuild)

File: recipes/jasper/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_MODULE_PATH ${CONAN_CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/source_subfolder/build/cmake/modules ${CMAKE_MODULE_PATH})
```
Comment (danimtb) [inline recipes/jasper/all/CMakeLists.txt]: Why is this needed?

Comment (SpaceIm) [reply recipes/jasper/all/CMakeLists.txt]: Because in its CMakeLists, Jasper use CMAKE_SOURCE_DIR instead of CMAKE_CURRENT_SOURCE_DIR
https://github.com/mdadams/jasper/blob/9aef6d91a82a8a6aecb575cbee57f74470603cc2/CMakeLists.txt#L6

Comment (danimtb) [reply recipes/jasper/all/CMakeLists.txt]: Thanks for the clarification!
